FNP 2020 - FinCausal/baseline/task1/task1_baseline.py

```diff
@@ -0,0 +1,524 @@
+
+dev_input_ids = pad_sequences(dev_input_ids, maxlen=MAX_LEN, dtype="long", 
+                          value=0, truncating="post", padding="post")
+
+print('\nDone.')
+
+# Create attention masks
+attention_masks = []
+dev_attention_masks = []
+
+# For each sentence...
+for sent in input_ids:
+    
+    # Create the attention mask.
+    #   - If a token ID is 0, then it's padding, set the mask to 0.
+    #   - If a token ID is > 0, then it's a real token, set the mask to 1.
+    att_mask = [int(token_id > 0) for token_id in sent]
+    
+    # Store the attention mask for this sentence.
+    attention_masks.append(att_mask)
+
+# For each sentence...
+for sent in dev_input_ids:
+    
+    # Create the attention mask.
+    #   - If a token ID is 0, then it's padding, set the mask to 0.
+    #   - If a token ID is > 0, then it's a real token, set the mask to 1.
+    att_mask = [int(token_id > 0) for token_id in sent]
+    
+    # Store the attention mask for this sentence.
+    dev_attention_masks.append(att_mask)
+    
+
+# Use 90% for training and 10% for validation.
+train_inputs = input_ids 
+validation_inputs = dev_input_ids
+train_labels = labels
+validation_labels = dev_labels
+
+# Do the same for the masks.
+train_masks =  attention_masks
+validation_masks = dev_attention_masks
+                                             
+# Convert all inputs and labels into torch tensors, the required datatype 
+# for our model.
+train_inputs = torch.tensor(train_inputs).to(torch.int64)
+validation_inputs = torch.tensor(validation_inputs).to(torch.int64)
+
+train_labels = torch.tensor(train_labels).to(torch.int64)
+validation_labels = torch.tensor(validation_labels).to(torch.int64)
+
+train_masks = torch.tensor(train_masks)
+validation_masks = torch.tensor(validation_masks)
+
+batch_size = 8
+
+# Create the DataLoader for our training set.
+train_data = TensorDataset(train_inputs, train_masks, train_labels)
+train_sampler = RandomSampler(train_data)
+train_dataloader = DataLoader(train_data, sampler=train_sampler, batch_size=batch_size)
+
+# Create the DataLoader for our validation set.
+validation_data = TensorDataset(validation_inputs, validation_masks, validation_labels)
+validation_sampler = SequentialSampler(validation_data)
+validation_dataloader = DataLoader(validation_data, sampler=validation_sampler, batch_size=batch_size)
+
+# Load BertForSequenceClassification, the pretrained BERT model with a single 
+# linear classification layer on top. 
+model = BertForSequenceClassification.from_pretrained(
+    "bert-base-uncased", # Use the 12-layer BERT model, with an uncased vocab.
+    num_labels = 2, # The number of output labels--2 for binary classification.
+                    # You can increase this for multi-class tasks.   
+    output_attentions = False, # Whether the model returns attentions weights.
+    output_hidden_states = False, # Whether the model returns all hidden-states.
+)
+
+# Tell pytorch to run this model on the GPU.
```

Comment:
Tell pytorch to run this model on the GPU if available -with cuda backend, CPU otherwise